mage_ai/orchestration/pipeline_scheduler.py

```diff
@@ -384,6 +389,51 @@ def build_tags(self, **kwargs):
             base_tags['hostname'] = HOSTNAME
         return merge_dict(kwargs, base_tags)
 
+    @safe_db_query
+    def __check_run_timeout(self) -> None:
+        pipeline_run_timeout = self.pipeline_run.pipeline_schedule.timeout
+
+        if self.pipeline_run.started_at and pipeline_run_timeout:
+            time_difference = datetime.now(tz=pytz.UTC).timestamp() - \
+                self.pipeline_run.started_at.timestamp()
+            if time_difference > int(pipeline_run_timeout):
+                self.logger.error(
+                    f'Pipeline run timed out after {int(time_difference)} seconds',
+                    **self.build_tags(),
+                )
+                stop_pipeline_run(
+                    self.pipeline_run,
+                    self.pipeline,
+                    status=PipelineRun.PipelineRunStatus.FAILED
+                )
+                return
```

Comment:
do you still proceed with scheduling logic in `schedule` method after `stop_pipeline_run`?